pkg/api/handlers/compat/images.go

```diff
@@ -255,6 +255,10 @@ func CreateImageFromImage(w http.ResponseWriter, r *http.Request) {
 	// without this early check this function would return 200 but reported error via body stream soon after
 	// it's better to let caller know early via HTTP status code that request cannot be processed
 	_, err := shortnames.Resolve(runtime.SystemContext(), fromImage)
+	if err != nil && shortnames.IsShortName(fromImage) {
+		fromImage = fmt.Sprintf("%s/%s", "docker.io", fromImage)
+		_, err = shortnames.Resolve(runtime.SystemContext(), fromImage)
+	}
```

Comment:
I was suggesting something like:
```
if err != nil && shortnames.IsShortName(fromImage) {
		names, err = shortnames.Resolve(runtime.SystemContext(), fromImage)
                if len(names) == 1 {
                      fromImage=names[0]
               } else {
		       fromImage = fmt.Sprintf("%s/%s", "docker.io", fromImage)
              }
}
```

Reply:
That way if I have a shortname mapping, Podman service will use it, if there is not one then fail over to docker.io.

Reply:
Oh yes, my bad; I will fix it.

Reply:
but if a software ask for ubi8, shortname.Resolve shouldn't fail in the first place, so it wouldn't go in that condition, no ?

Reply:
I am not sure what you are asking.  shortnames.IsShortName("ubi8") should return true.
Then I would want to pull registry.access.redhat.com/ubi8 not docker.io/ubi8.

Reply:
Yes, but:
```
_, err := shortnames.Resolve(runtime.SystemContext(), fromImage)
```

will return nil in err if fromImage is a short name alias, since it can be resolved.
If there is no error, fromImage will not be modified.

Reply:
Your right I missed that
LGTM

Reply:
This is ignoring errors in short-name resolution which I think is something we cannot do.  Let's go one step back and first define the problem we want to solve.  This looks like a way to big shotgun to me.